components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1000,4 +1034,108 @@ private ServiceProvider doAddApplication(ServiceProvider serviceProvider, String
         }
         return serviceProvider;
     }
+
+    /**
+     * Unmarshal service provider.
+     *
+     * @param content      string
+     * @param fileName     file name of the service provider
+     * @param tenantDomain tenant domain name
+     * @return Service Provider
+     * @throws IdentityApplicationManagementException Identity Application Management Exception
+     */
+    private ServiceProvider unmarshalSP(String content, String fileName, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        if (content == null || StringUtils.isEmpty(content)) {
+            throw new IdentityApplicationManagementException(String.format("Empty Service Provider file %s upload by" +
+                    " tenant: %s", fileName, tenantDomain));
+        }
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(ServiceProvider.class);
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+            return (ServiceProvider) unmarshaller.unmarshal(new ByteArrayInputStream(
+                    content.getBytes(StandardCharsets.UTF_8)));
+
+        } catch (JAXBException e) {
+            throw new IdentityApplicationManagementException(String.format("Error in reading Service Provider file %s" +
+                    " upload by" +
+                    " tenant: %s", fileName, tenantDomain));
+        }
+    }
+
+    /**
+     * Marshal Service provider to string.
+     *
+     * @param serviceProvider service provider to be marshaled
+     * @param tenantDomain    tenant domain
+     * @return string
+     * @throws IdentityApplicationManagementException Identity Application Management Exception
+     */
+    private String marshalSP(ServiceProvider serviceProvider, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(ServiceProvider.class);
+            Marshaller marshaller = jaxbContext.createMarshaller();
+            DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
```

Comment:
There was a Util to get DBF with the required security properties. Please check and update if required